var/spack/repos/builtin/packages/py-pip/package.py

```diff
@@ -57,8 +66,15 @@ def install(self, spec, prefix):
         # already installed. We get around this by using a pre-built wheel to install
         # itself, see:
         # https://discuss.python.org/t/bootstrapping-a-specific-version-of-pip/12306
+        # On Windows for newer versions of pip, you must bootstrap pip first. In order
+        # to achieve this, use the pip.pyz zipapp version of pip to bootstrap the
+        # wheel install.
         whl = self.stage.archive_file
-        args = [os.path.join(whl, "pip")] + std_pip_args + ["--prefix=" + prefix, whl]
+        args = std_pip_args + ["--prefix=" + prefix, whl]
+        if sys.platform == "win32":
+            args.insert(0, os.path.join(self.stage.source_path, "pip.pyz"))
+        else:
+            args.insert(0, os.path.join(whl, "pip"))
```

Comment:
This looks great but can you rearrange it so all the Windows discussion is isolated to the Windows branch of the `if`: like

```
whl = self.stage.archive_file
if sys.platform == "win32":
    # On Windows for newer versions of pip...
    args = []
else:
    args = []
args  += std_pip_args + ["--prefix=" + prefix, whl]
```

?